tests/integration/src/test/java/org/apache/pulsar/tests/integration/containers/PulsarInitMetadataContainer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.tests.integration.containers;
+
+import java.io.IOException;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+
+/**
+ * Initialize the Pulsar metadata
+ */
+@Slf4j
+public class PulsarInitMetadataContainer extends GenericContainer<PulsarInitMetadataContainer> {
+
+    public static final String NAME = "init-metadata";
+
+    private final String clusterName;
+    private final String metadataStoreUrl;
+    private final String configurationMetadataStoreUrl;
+    private final String brokerHostname;
+
+    public PulsarInitMetadataContainer(Network network,
+                                       String clusterName,
+                                       String metadataStoreUrl,
+                                       String configurationMetadataStoreUrl,
+                                       String brokerHostname) {
+        this.clusterName = clusterName;
+        this.metadataStoreUrl = metadataStoreUrl;
+        this.configurationMetadataStoreUrl = configurationMetadataStoreUrl;
+        this.brokerHostname = brokerHostname;
+        setDockerImageName(PulsarContainer.DEFAULT_IMAGE_NAME);
+        withNetwork(network);
+
+        setCommand("sleep 1000000");
```

Comment:
Do we need this sleep command?

Reply:
Yes, this spins up a container just idle. The command to initialize the cluster is then `execd` into the container, to retrieve the exit code.